### PR TITLE
Edit variation price

### PR DESCRIPTION
**Pull request by 0nko**
This PR refactors the product pricing screen by decoupling it from the product detail. There is a new `ProductPricingViewModel`, which takes over all price-editing logic from `ProductDetailViewModel`. The pricing screen is now independent and can be used for variation price editing.

The price screen UI should be the same. The same holds for product price editing behaviour. 

| Light | Dark |
|-|-|
| image | image |

**To test:**

_Product details_
1. Select a simple product and go to the product details screen
2. Tap on the Price property and open the pricing screen
3. Smoke test price editing to make sure the functionality is correct
4. Make sure the Done button is visible only when there is a change
5. Check that the end sale date cannot be > start sale date
6. Check that the sale price cannot be > regular price
7. Check that the discard dialog appears if there is a change and back button is pressed
8. Make some changes, tap one the Done button
7. Verify that the changes are reflected in the product details

_Variation details_
1. Select a variable product
2. Tap on Variations property
3. Select a variation and open variation detail screen
4. Tap on the Price property
5. Notice the tax section is not visible
6. Smoke test price editing to make sure the functionality is correct
7. Make some changes, tap one the Done button
8. Verify that the changes are reflected in the product details

_Unit tests_
1. Run tests in `ProductPricingViewModelTest`
2. Verify all tests pass

### CODE REVIEW

**Comment by 0nko** [comment]
@ThomazFB I discovered a race condition when the price result was returned to the calling fragment. There is a new way of returning results in the latest Navigation component library version (2.3), so I updated it and refactored the code to use it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductPricingFragment.kt [hunk 1]
```diff
@@ -337,16 +373,19 @@ class ProductPricingFragment : BaseProductFragment(), ProductInventorySelectorDi
...
     }
 
     override fun onRequestAllowBackPress(): Boolean {
-        return viewModel.onBackButtonClicked(ExitPricing())
+        return if (viewModel.event.value == Exit) {
+            true
+        } else {
+            viewModel.onExit()
+            false
+        }
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductPricingFragment.kt:389]: np: just a suggestion, nothing essential, but if you want, you can take advantage of the generated boolean of the comparison itself and hide the explicit Boolean returns and use the `also` for the `false` scenario where the `onExit()` needs to be called.

```suggestion
        return (viewModel.event.value == Exit)
        .also { if(it.not()) viewModel.onExit() }
```

**Comment by ThomazFB** [comment]
Hey @0nko, great work! I've left a single code suggestion, but absolutely nothing essential, tests are passing and the pricing modification feature is working just fine! 

One additional thing, I've noticed that the target branch to merge this PR is the `tests/fix-failing-product-detail-tests`, is that the correct target branch to receive the merge? I noticed that the previous PR I approved right here https://github.com/woocommerce/woocommerce-android/pull/2674 was exactly from this branch into the `issue/2193-edit-variation`. Have I made a mistake with the review order? 😰 

Nevertheless, the implementation looks great! LGTM :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductPricingFragment.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductPricingFragment.kt]: Thanks 👍

**Comment by 0nko** [comment]
Thanks for the review, @ThomazFB. I've stacked multiple PRs so that I could continue working while they're being reviewed. After you merge a PR and delete the branch, a PR referencing that branch will switch automatically to the target branch.

**Pull request merged**